hack/ginkgo-e2e.sh

```diff
@@ -169,6 +169,7 @@ fi
 "${program[@]}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
   "${auth_config[@]:+${auth_config[@]}}" \
   --ginkgo.flakeAttempts="${FLAKE_ATTEMPTS}" \
+  --ginkgo.timeout="150m" \
```

Comment:
this option is available from `GinkgoConfiguration()` ,

https://github.com/onsi/ginkgo/blob/ff12cee51ec66a5d644fd275a304095163f6fc18/types/config.go#L31

 since we are already are modifying some ginkgo values I think we should set this variable directly for backwards compatibility
https://github.com/kubernetes/kubernetes/pull/109111/commits/64d29d99607b520ab3390a25406f8c940acaa6fe#r857104476

also, I will start with the same value of 24h, and set a TODO to refine later, WDYT @pohly

Reply:
I agree that it would be better if we could ensure that e2e.test uses the traditional 24h hour timeout and then not modify any of the places where the PR currently has to change the -ginkgo.timeout parameter. Otherwise this PR has another "Action required: ensure that you explicitly increase the timeout" - not nice.

My own idea over the weekend (untested) was to reconfigure the timeout during init:

```
func init() {
    flag.Set("-ginkgo.timeout", "24h")
}
```

If `ginkgo ./test/e2e` then doesn't pass its own default timeout to the `e2e.test` binary, we should run with the 24h timeout. I'm a bit worried that the timeout becomes hardcoded when using the approach via `GinkgoConfiguration()`. This needs further exploration.

One thing that I learned from playing with Ginkgo v2 in the PMEM-CSI CI (Jenkins) is that it pays of to run ginkgo with a timeout that is a bit shorter than the timeout for the entire job: when ginkgo times out itself, it still cleans up and writes the JUnit file. If it gets interrupted by Jenkins, it doesn't do that.

I don't know how Prow kills jobs. It might behave differently, for example use a signal that gets caught by ginkgo. Again, this needs some exploration.

Reply:
@pohly bear in mind that the gingo author @onsi offered his help, so we can try to get his opinion about things we don't have clear

Reply:
👋 @pohly opened a few issues over on the Ginkgo repo earlier this week and I've also been going back and forth a bit with @chendave - I'd love to step back and do a quick zoom call with you all (as I'd mentioned to @pohly over slack) about the issues @pohly opened and about the migration in general to see if there's anything I can do/adjust in Ginkgo to help.

would y'all be up for doing that?

As a quick example - while you can change values in `GinkgoConfiguration()` not all changes are supported cleanly and which ones are/aren't isn't well documented.  So this issue is leading me to want to add some support in Ginkgo to make it clear to you which configuration changes are safe to make at runtime vs which should be done via flags.

In this particular case I would not adjust `Timeout` via `GinkgoConfiguration()`.  It'll work for the current test suite but when the CLI runs multiple suites it sets up the timeout for each suite it runs to ensure that the total runtime of all suites is within the allowed timeout.  If you override it on a per-suite basis you'll lose the cross-suite timeout.

So, a topic on a zoom call might be to explore timeouts specifically (which was one of the issues @pohly opened) but also potentially configuration more broadly.  Thoughts?  I'm free most of this coming week.

Reply:
The issue for timeouts is https://github.com/onsi/ginkgo/issues/969

> when the CLI runs multiple suites it sets up the timeout for each suite it runs to ensure that the total runtime of all suites is within the allowed timeout

That is about `ginkgo ./test-suite-a ./test-suite-b` and executing them sequentially, right? How does it then decide how long each suite is allowed to run? Each gets half of the overall timeout? This is less relevant for us because everything is in a single test/e2e for Kubernetes. I don't think many users will run it together with other tests.

Reply:
My idea with overriding the default inside the test suite doesn't work because the `ginkgo` command always passes a `-ginkgo.timeout` parameter, even if the `ginkgo` command itself wasn't invoked with `--timeout`.

What we would need is a way for the suite to define a different default timeout and then have the `ginkgo` command pick that up when running it.

Reply:
Perhaps we should simply accept that now `ginkgo -timeout=24h` has to be used. Users already must change from the v1 CLI to the v2 CLI, so also telling them to add one parameter while doing so isn't that much harder.

Reply:
@chendave : I suggest the following wording:

```
ACTION REQUIRED: the e2e test suite now uses ginkgo v2. When running it via the ginkgo CLI, the v2 release must be used and `-timeout=24h` (or some other, suitable value) must be passed because the default timeout was reduced from 24h to 1h. When running it via `go test`, the corresponding `-args` parameter is `-ginkgo.timeout=24h`.
```

Reply:
> That is about ginkgo ./test-suite-a ./test-suite-b and executing them sequentially, right? How does it then decide how long each suite is allowed to run? Each gets half of the overall timeout? This is less relevant for us because everything is in a single test/e2e for Kubernetes. I don't think many users will run it together with other tests.

It gives each sequential suite all the remaining time.  but fair enough, it's not relevant to you all.  however - yes - as you are seeing the Ginkgo cli will always set the timeout.  Given the new default I don't think there's much else owe can do here other than mark it as an `ACTION REQUIRED`.  Hopefully there Han't been too many of those??

Reply:
We don't know how many people will be affected. The e2e.test binary is used for Kubernetes conformance testing, but I think many people do that via sonobuoy without directly invoking it themselves.

Let's go with "ACTION REQUIRED".

Reply:
okay, I have updated the release note, and will add one line of note in the source code as well.

@onsi thanks for offering those great suggestion!  I am a little busy on other things this week and just found some time to address some of comments.
so far,  I think we almost there. Only one issue still need to fix is that there is a custom reporter which named as `ProgressReporter`, here is the source for that [ProgressReporter](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/reporters/progress.go), this reporter need some real-time reporting capabilities, I am a little worried whether this could be implemented by `ReportBeforeEach` and `ReportAfterEach`.

I am going to have a try, but if you have a quick answer, pls kindly left your comment here.

> quick zoom call with you all 

thanks for the proposal, I am going to have a try first, might ping you later.